src/shell.js

```diff
@@ -188,6 +169,28 @@ if (ENVIRONMENT_IS_NODE) {
   };
 
   Module['inspect'] = function () { return '[Emscripten Module object]'; };
+
+#if USE_PTHREADS
+  var nodeWorkerThreads;
+  try {
+    nodeWorkerThreads = require('worker_threads');
+  } catch (e) {
+    console.error('The "worker_threads" module is not supported in this node.js build - perhaps a newer version is needed?');
+    throw e;
+  }
+  Worker = nodeWorkerThreads.Worker;
+
+  // Polyfill the performance object, which emscripten pthreads support
+  // depends on for good timing.
```

Comment:
does it depend on the good resolution of performance.now?

Reply:
I think the assumption back then was that we want maximal precision for futex etc. timing.

(However meanwhile the web has decreased performance.now precision so it's possible that's changed?)

Reply:
Node.js supports performance as `require('perf_hooks').performance` … it’s technically still experimental, but predates Workers and is unlikely to be removed or changed, esp. since it’s there for compatibility with the Web API.

Reply:
Thanks, I'll use that.